arlo_server/all_routes/election/__element__.py

```diff
@@ -0,0 +1,91 @@
+from flask import jsonify, Response, request
+from jsonschema import validate
+
+from arlo_server import app, db
+from arlo_server.auth import require_audit_admin_for_organization
+from arlo_server.models import Election
+
+ELECTION_NAME_KEY = "electionName"
+ELECTION_ONLINE_KEY = "online"
+ELECTION_RANDOM_SEED_KEY = "randomSeed"
+ELECTION_RISK_LIMIT_KEY = "riskLimit"
+ELECTION_STATE_KEY = "state"
+
+PUT_ELECTION_REQUEST_SCHEMA = {
+    "type": "object",
+    "properties": {
+        ELECTION_NAME_KEY: {"type": "string"},
+        ELECTION_ONLINE_KEY: {"type": "boolean"},
+        ELECTION_RANDOM_SEED_KEY: {"type": "string"},
+        ELECTION_RISK_LIMIT_KEY: {"type": "integer"},
+        ELECTION_STATE_KEY: {"type": "string"},
+    },
+    "additionalProperties": False,
+}
+
+GET_ELECTION_RESPONSE_SCHEMA = {
+    "type": "object",
+    "properties": {
+        "election": {
+            "type": "object",
+            "properties": {
+                ELECTION_NAME_KEY: {"type": "string"},
+                ELECTION_ONLINE_KEY: {"type": "boolean"},
+                ELECTION_RANDOM_SEED_KEY: {"type": "string"},
+                ELECTION_RISK_LIMIT_KEY: {"type": "integer"},
+                ELECTION_STATE_KEY: {"type": "string"},
+            },
+            "additionalProperties": False,
+        },
+    },
+    "additionalProperties": False,
+}
+
+
+@app.route("/election/<election_id>", methods=["GET"])
+def get_election(election_id: str) -> Response:
+    election = Election.query.filter_by(id=election_id).one()
+    require_audit_admin_for_organization(election.organization_id)
+
+    response_data = {
+        "election": {
+            ELECTION_NAME_KEY: election.name,
+            ELECTION_ONLINE_KEY: election.online,
+            ELECTION_RANDOM_SEED_KEY: election.random_seed,
+            ELECTION_RISK_LIMIT_KEY: election.risk_limit,
+            ELECTION_STATE_KEY: election.state,
+        }
+    }
+
+    validate(schema=GET_ELECTION_RESPONSE_SCHEMA, instance=response_data)
```

Comment:
What's the thinking behind validating this? Should already be valid in the db, right?

Reply:
It's not so much validating the types of the fields as forcing us to have some sort of defined structure for our responses. At Square, we used [Protocol Buffers](https://en.wikipedia.org/wiki/Protocol_Buffers), which I found to be a nice way to define our APIs. It was awesome to have files that clearly documented our request and response data structures in a way that could not get out of sync with the actual implementation. That's my main problem with the `.md` files we have for API docs. They're out of date practically instantly.

This at least ensures that we adhere to the schema we say we do.

Reply:
I definitely like the idea of having schemas to document our request/response data structures. I agree that it's way better than written docs, especially if they are tied to the code.

I did a quick google and found this: https://pypi.org/project/jsonschema-typed/. Could that help?

Reply:
Yeah, something like that might be a good way to go. Let's explore in a follow-up.